src/openms/include/OpenMS/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.h

```diff
@@ -239,20 +239,60 @@ namespace OpenMS
           peak_apex = (best_left + best_right) / 2.0;
         }
       }
-      picked_chroms[chr_idx][peak_idx].setIntensity(0.0);
 
-      // Remove other, overlapping, picked peaks (in this and other
-      // chromatograms) and then ensure that at least one peak is set to zero
-      // (the currently best peak).
-      remove_overlapping_features(picked_chroms, best_left, best_right);
+      std::vector< double > left_edges;
+      std::vector< double > right_edges;
+      double min_left = best_left;
+      double max_right = best_right;
+      if (use_consensus_)
+      {
+        // Remove other, overlapping, picked peaks (in this and other
+        // chromatograms) and then ensure that at least one peak is set to zero
+        // (the currently best peak).
+        remove_overlapping_features(picked_chroms, best_left, best_right);
+      }
+      else
+      {
+        // Pick the most intense peak for each chromatogram and use this for
+        // the current peak. Note that we will only set the most intense peak
+        // to zero, so if there are two peaks for some transitions, we will get
+        // to them later.
```

Comment:
Why is it necessary to retain information about the largest peak in the transition group?  From my understanding of the rest of the code, it is just going to use the `local_left` and `local_right` as derived from the `left_edges` and `right_edges` array, it would not be needed to keep track of the largest peak in the transition group.  Am I missing something here?

Reply:
So this code works as follows:

1. Pick peaks in all chromatograms
2. Iterate over all peaks in all chromatograms by width / intensity
  2.1 Identify current peak (largest / widest overall)
  2.2 Identify peaks in other chromatograms that fall within current peak
  2.3 Pick one suitable peak in each chromatogram

Note that we run into problems if in step 2.2 there is no peak or there is more than one peak